libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -646,15 +646,30 @@ void NavEKF3_core::readGpsData()
 
             // see if we can get an origin from the frontend
             if (!validOrigin && frontend->common_origin_valid) {
-                setOrigin(frontend->common_EKF_origin);
+
+                if (!setOrigin(frontend->common_EKF_origin)) {
+                    // rejects setting the origin more than once and sets an error
```

Comment:
```suggestion
                    // rejects setting the origin more than once and sets an error
```

Not it doesn't - similarly elsewhere

Reply:
fixed :)